apis/Google.Cloud.Firestore/Google.Cloud.Firestore.Tests/QueryTest.cs

```diff
@@ -990,7 +992,6 @@ public void Equality_Where()
             EqualityTester.AssertEqual(query,
                 equal: new[] {
                     col.WhereEqualTo("field", 10), // Same value
-                    col.WhereEqualTo("field", 10L), // Same value after serialization
```

Comment:
Why was this line removed? (It's much easier to spot now that there are fewer changes in this file.)

Reply:
Earlier we used to convert object type to value type(serialization) right after it is passed into the `Where*` methods. Now we have delegated it to later (while converting to proto in `Filter.ToProto()`). So this statement will fail now.

Reply:
Hmm. That's a *bit* of a concern in terms of compatibility; I hadn't considered that before. For the moment, please highlight that in a separate test, as a record of the change. I'd like to think about that more carefully before we merge - it's just possible that users could be relying on the current behavior.

Reply:
I was considering that we *could* convert to a proto when checking for equality - but that also risks throwing an exception, if serialization fails. (Basically, changing the serialization timing knocks things somewhat.)

Another option to consider: make `Query.Filter` do the serialization. That's the earliest point at which we have all the required information - and it makes it fail early if serialization fails, as well as helping with equality operations. I haven't tried this though - I suspect it would add significant complexity.

Reply:
Next thought: serialize on a call to Query.Where(Filter), but discard the results - so we keep the same structure that's in the PR, but we're assured that serialization will work. We can then *also* serialize on GetHashCode and Equals, so we keep compatibility with previous code. It's slightly more expensive, but I don't expect filters to actually be that large.